docs-api/frontity-packages/features-packages/wp-comments.md

```diff
@@ -0,0 +1,226 @@
+---
+description: API reference of `@frontity/wp-comments` package
+---
+
+# `@frontity/wp-comments`
+
+Comments package that adds integration for WordPress native comments.
+
+<!-- toc -->
+
+- [Installation](#installation)
+- [Settings](#settings)
+  - [In WordPress](#in-wordpress)
+  - [In Frontity](#in-frontity)
+- [API Reference](#api-reference)
+  - [Handlers](#handlers)
+    - [`@comments/:id`](#commentsid)
+  - [State](#state)
+    - [`state.comments.forms[]`](#state-comments-forms)
+  - [Actions](#actions)
+    - [`actions.comments.updateFields()`](#actions-comments-updatefields)
+    - [`actions.comments.submit()`](#actions-comments-submit)
+- [Demo](#demo)
+
+<!-- tocstop -->
+
+## Installation
+
+Add the `wp-comments` package to your project:
+
+```bash
+npm i @frontity/wp-comments
+```
+
+## Settings
+
+### In WordPress
+
+In order to use this package, you will need to add a single line of configuration to your Wordpress installation:
+
+```php
+add_filter( 'rest_allow_anonymous_comments', '__return_true' );
+```
+
+YYou can add that directly in your theme's `functions.php` file or use a [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin.
+
+### In Frontity
+
+This package doesn't have any configuration. It just needs to be added to the `packages` array in `frontity.settings.js`.
+
+**`frontity.settings.js`**
+
+```js
+export default {
+  packages: ["@frontity/wp-comments"],
+};
+```
+
+## API Reference
+
+### Handlers
+
+#### `@comments/:id`
+
+This is a `wp-source` handler for fetching comments from a specific post using its ID. For example, to fetch all comments that belong to the post with ID 60 you would do:
+
+```
+await actions.source.fetch("@comments/60");
+```
+
+This would fetch all comments associated with that post and populate a data object inside `state.source.data` with a tree structure of comments and replies, sorted by date (most recent first).
+
+To access the fetched comments you could use something similar to this example:
+
+```js
+const Comments = connect(({ postId, state }) => {
+  // Get comments from state.
+  const data = state.source.get(`@comments/${postId}`);
+
+  // Utility to render comments and replies recursively.
+  const renderComments = (items) =>
+    items.map(({ id, children }) => (
+      <Comment id={id}>
+        {/* Render replies */}
+        {children && renderComments(children)}
+      </Comment>
+    ));
+
+  // Render comments if data is ready.
+  return data.isReady ? renderComments(data.items) : null;
+});
+```
+
+### State
+
+#### `state.comments.forms[]`
+
+The `wp-comments` package stores in `state.comments.forms` a map of objects by post ID, representing each one a comment form. These objects are intended to be used as the state of React `<form>` components and contain the input values as well as the submission status. They have the following properties:
```

Comment:
@juanmaguitar TBH I struggled to understand this. Having spent some time reading and re-reading it I think I understand it to be related to form submission. Maybe this could be made clearer?

Reply:
Thanks for the feedback. Yes, it's a bit tricky to be understood. Let me try to improve the explanation